src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,461 @@
+                ? Mode.MODULE_CULLED
+                : Mode.NORMAL
+            );
+        } else {
+            // Normal path
+            (uint256 totalAssets, uint256 targetAssets) = ilks[ilk_].plan.calcSupplies(availableAssets);
+
+            if (targetAssets > totalAssets) {
+                _wind(ilk_, pool, targetAssets - totalAssets);
+            } else if (targetAssets < totalAssets) {
+                _unwind(
+                    ilk_,
+                    pool,
+                    totalAssets - targetAssets,
+                    availableAssets,
+                    Mode.NORMAL
+                );
+            }
+        }
+    }
+
+    // --- Collect Interest ---
+    function reap(bytes32 ilk_) external {
+        D3MPoolLike pool = ilks[ilk_].pool;
+
+        require(vat.live() == 1, "DssDirectDepositHub/no-reap-during-shutdown");
+        require(live == 1, "DssDirectDepositHub/no-reap-during-cage");
+
+        uint256 assetBalance = pool.assetBalance();
+        (, uint256 daiDebt) = vat.urns(ilk_, address(pool));
+        if (assetBalance > daiDebt) {
+            uint256 fees = assetBalance - daiDebt;
+            uint256 availableAssets = pool.maxWithdraw();
+            if (fees > availableAssets) {
+                fees = availableAssets;
+            }
+            pool.withdraw(fees);
+            daiJoin.join(vow, fees);
+            emit Reap(ilk_, fees);
+        }
+    }
+
+    // --- Allow DAI holders to exit during global settlement ---
+    function exit(bytes32 ilk_, address usr, uint256 wad) external {
+        require(wad <= 2 ** 255, "DssDirectDepositHub/overflow");
+        vat.slip(ilk_, msg.sender, -int256(wad));
+        D3MPoolLike pool = ilks[ilk_].pool;
+        require(pool.transferShares(usr, pool.convertToShares(wad)), "DssDirectDepositHub/failed-transfer");
+        emit Exit(ilk_, usr, wad);
+    }
+
+    // --- Shutdown ---
+    function cage(bytes32 ilk_) external {
+        require(vat.live() == 1, "DssDirectDepositHub/no-cage-during-shutdown");
+
+        D3MPoolLike pool = ilks[ilk_].pool;
+
+        // Can shut joins down if we are authed
+        // or if the interest rate strategy changes
+        // or if the main module is caged
+        require(
+            wards[msg.sender] == 1 ||
+            live == 0 ||
+            !pool.validTarget()
+        , "DssDirectDepositHub/not-authorized");
+
+        pool.cage();
+        ilks[ilk_].tic = block.timestamp;
+        emit Cage(ilk_);
+    }
+
+    function cage() external {
+        require(wards[msg.sender] == 1 , "DssDirectDepositHub/not-authorized");
+        require(vat.live() == 1, "DssDirectDepositHub/no-cage-during-shutdown");
+
+        live = 0;
+        emit Cage();
+    }
+
+    // --- Write-off ---
+    function cull(bytes32 ilk_) external {
+        require(vat.live() == 1, "DssDirectDepositHub/no-cull-during-shutdown");
+        require(live == 0, "DssDirectDepositHub/live");
+        Ilk storage ilk = ilks[ilk_];
```

Comment:
No that we also have the `plan` we shouldn't be calling the whole struct here neither.